lang/codegen/src/generator/common.rs

```diff
@@ -195,4 +247,76 @@ pub(crate) mod casper {
             }
         }
     }
+
+    pub fn serialize_enum(enum_ident: &Ident, variants: &[Variant]) -> TokenStream {
+        let from_bytes_code = enum_from_bytes(enum_ident, variants);
+        let to_bytes_code = enum_to_bytes(enum_ident, variants);
+
+        quote! {
+            #[cfg(any(feature = "casper", feature = "casper-livenet"))]
+            impl odra::casper::casper_types::bytesrepr::FromBytes for #enum_ident {
+                #from_bytes_code
+            }
+
+            #[cfg(any(feature = "casper", feature = "casper-livenet"))]
+            impl odra::casper::casper_types::bytesrepr::ToBytes for #enum_ident {
+                #to_bytes_code
+            }
+
+            #[cfg(any(feature = "casper", feature = "casper-livenet"))]
+            impl odra::casper::casper_types::CLTyped for #enum_ident {
+                fn cl_type() -> odra::casper::casper_types::CLType {
+                    odra::casper::casper_types::CLType::U8
+                }
+            }
+        }
+    }
+
+    fn enum_from_bytes(enum_ident: &Ident, variants: &[Variant]) -> TokenStream {
+        let append_bytes = variants
+            .iter()
+            .map(|variant| {
+                let ident = &variant.ident;
+                quote! {
+                    x if x == #enum_ident::#ident as u8 => std::result::Result::Ok((#enum_ident::#ident, bytes))
+                }
+            })
+            .collect::<Punctuated<TokenStream, Comma>>();
+
+        quote! {
+            fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), odra::casper::casper_types::bytesrepr::Error> {
+                let (variant, bytes): (u8, _) = odra::casper::casper_types::bytesrepr::FromBytes::from_bytes(bytes)?;
+                match variant {
+                    #append_bytes,
+                    _ => std::result::Result::Err(odra::casper::casper_types::bytesrepr::Error::Formatting),
+                }
+            }
+        }
+    }
+
+    fn enum_to_bytes(enum_ident: &Ident, variants: &[Variant]) -> TokenStream {
+        let append_bytes = variants
+            .iter()
+            .map(|variant| {
+                let ident = &variant.ident;
+                quote!(#enum_ident::#ident => #enum_ident::#ident as u8,)
+            })
+            .collect::<TokenStream>();
+
+        quote! {
+            fn serialized_length(&self) -> usize {
+                1
+            }
+
+            fn to_bytes(&self) -> Result<Vec<u8>, odra::casper::casper_types::bytesrepr::Error> {
+                let mut vec = std::vec::Vec::with_capacity(self.serialized_length());
```

Comment:
All elements of the enum are u32, so you should be able to write it in a generic way:
```rust
(self.clone() as u32).to_bytes()
```